src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,263 @@
+import { html, TemplateResult, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import componentStyles from './outline-modal.css.lit';
+import { OutlineElement } from '../outline-element/outline-element';
+import { ifDefined } from 'lit/directives/if-defined';
+
+export type ModalSize = 'small' | 'medium' | 'full-screen';
+
+export const modalSizes: ModalSize[] = ['small', 'medium', 'full-screen'];
+
+// See https://stackoverflow.com/questions/1599660/which-html-elements-can-receive-focus.
+// @todo make this re-usable across components?
+const focusableElementSelector = `
+  a[href]:not([tabindex="-1"]),
+  area[href]:not([tabindex="-1"]),
+  input:not([disabled]):not([tabindex="-1"]),
+  select:not([disabled]):not([tabindex="-1"]),
+  textarea:not([disabled]):not([tabindex="-1"]),
+  button:not([disabled]):not([tabindex="-1"]),
+  iframe:not([tabindex="-1"]),
+  [tabindex]:not([tabindex="-1"]),
+  [contentEditable=true]:not([tabindex="-1"])
+`;
+
+/**
+ * The Outline Modal component
+ * @element outline-modal
+ * @slot default - The modal contents
+ * @slot outline-modal--trigger - The trigger for the modal
+ * @slot outline-modal--header - The header in the modal
+ * @slot outline-modal--accessibility-description - The accessibility description which is used by screen readers.
+ */
+@customElement('outline-modal')
+export class OutlineModal extends OutlineElement {
+  static styles: CSSResultGroup = [componentStyles];
+
+  render(): TemplateResult {
+    return html`
+      <div
+        id="trigger"
+        tabindex="0"
+        @click="${this._handleModalTrigger}"
+        @keydown="${this._handleModalTrigger}"
+      >
+        <slot name="outline-modal--trigger"></slot>
+      </div>
+      ${this._overlayTemplate()}
+    `;
+  }
+
+  @property({ attribute: false })
+  isOpen = false;
+
+  @property({ type: String })
+  size?: ModalSize = 'medium';
+
+  private _overlayTemplate(): TemplateResult {
+    let template = html``;
+
+    if (this.isOpen) {
+      template = html`
+        <div
+          id="overlay"
+          tabindex="-1"
+          class="${this.size}"
+          @click="${this._handleModalClose}"
+          @keydown="${this._handleOverlayKeyup}"
```

Comment:
This needs to be updated. When I switched from `keyup` to `keydown`, I forgot to rename the method. :(

I'll try to tackle that on Monday.